frame/nis/src/lib.rs

```diff
@@ -0,0 +1,931 @@
+{
+	fn convert_back(a: A::Type) -> Perquintill {
+		Perquintill::from_rational(a, A::get())
+	}
+}
+
+pub struct NoCounterpart<T>(sp_std::marker::PhantomData<T>);
+impl<T> FungibleInspect<T> for NoCounterpart<T> {
+	type Balance = u32;
+	fn total_issuance() -> u32 {
+		0
+	}
+	fn minimum_balance() -> u32 {
+		0
+	}
+	fn balance(_who: &T) -> u32 {
+		0
+	}
+	fn reducible_balance(_who: &T, _keep_alive: bool) -> u32 {
+		0
+	}
+	fn can_deposit(
+		_who: &T,
+		_amount: u32,
+		_mint: bool,
+	) -> frame_support::traits::tokens::DepositConsequence {
+		frame_support::traits::tokens::DepositConsequence::Success
+	}
+	fn can_withdraw(
+		_who: &T,
+		_amount: u32,
+	) -> frame_support::traits::tokens::WithdrawConsequence<u32> {
+		frame_support::traits::tokens::WithdrawConsequence::Success
+	}
+}
+impl<T> FungibleMutate<T> for NoCounterpart<T> {
+	fn mint_into(_who: &T, _amount: u32) -> DispatchResult {
+		Ok(())
+	}
+	fn burn_from(_who: &T, _amount: u32) -> Result<u32, DispatchError> {
+		Ok(0)
+	}
+}
+impl<T> Convert<Perquintill, u32> for NoCounterpart<T> {
+	fn convert(_: Perquintill) -> u32 {
+		0
+	}
+}
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::{FungibleInspect, FungibleMutate};
+	pub use crate::weights::WeightInfo;
+	use frame_support::{
+		pallet_prelude::*,
+		traits::{
+			nonfungible::{Inspect as NonfungibleInspect, Transfer as NonfungibleTransfer},
+			Currency, Defensive, DefensiveSaturating,
+			ExistenceRequirement::AllowDeath,
+			OnUnbalanced, ReservableCurrency,
+		},
+		PalletId,
+	};
+	use frame_system::pallet_prelude::*;
+	use sp_arithmetic::{PerThing, Perquintill};
+	use sp_runtime::{
+		traits::{AccountIdConversion, Bounded, Convert, ConvertBack, Saturating, Zero},
+		TokenError,
+	};
+	use sp_std::prelude::*;
+
+	type BalanceOf<T> =
+		<<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+	type PositiveImbalanceOf<T> = <<T as Config>::Currency as Currency<
+		<T as frame_system::Config>::AccountId,
+	>>::PositiveImbalance;
+	type ReceiptRecordOf<T> = ReceiptRecord<
+		<T as frame_system::Config>::AccountId,
+		<T as frame_system::Config>::BlockNumber,
+	>;
+	type IssuanceInfoOf<T> = IssuanceInfo<BalanceOf<T>>;
+	type SummaryRecordOf<T> = SummaryRecord<<T as frame_system::Config>::BlockNumber>;
+	type BidOf<T> = Bid<BalanceOf<T>, <T as frame_system::Config>::AccountId>;
+	type QueueTotalsTypeOf<T> = BoundedVec<(u32, BalanceOf<T>), <T as Config>::QueueCount>;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// Information on runtime weights.
+		type WeightInfo: WeightInfo;
+
+		/// Overarching event type.
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+		/// The treasury's pallet id, used for deriving its sovereign account ID.
+		#[pallet::constant]
+		type PalletId: Get<PalletId>;
+
+		/// Currency type that this works on.
+		type Currency: ReservableCurrency<Self::AccountId, Balance = Self::CurrencyBalance>;
```

Comment:
I think there's no need to use that trait specifically in this pallet and force it to handle the naming. If naming is needed by the runtime then it can be done through an adapter struct at the pallet config level. I'll provide such an adapter in this PR.